testing/src/test/java/org/motechproject/nms/testing/it/api/OpsControllerBundleIT.java

```diff
@@ -368,4 +407,86 @@ private void initializeLocationData() {
+
+       subscriptionService.createSubscription(subscriberIVR.getCallingNumber(), rh.kannadaLanguage(), rh.karnatakaCircle(),
+                sh.pregnancyPack(), SubscriptionOrigin.IVR);
+
+        Subscriber subscriberMCTS = subscriberDataService.create(new Subscriber(2000000000L));
+        subscriberMCTS.setLastMenstrualPeriod(DateTime.now().plusWeeks(70));
+        subscriberMCTS = subscriberDataService.update(subscriberMCTS);
+        subscriptionService.createSubscription(subscriberMCTS.getCallingNumber(), rh.kannadaLanguage(), rh.karnatakaCircle(),
+                sh.pregnancyPack(), SubscriptionOrigin.MCTS_IMPORT);
+        transactionManager.commit(status);
+    }
+
+
+    public void testifSubscriberDectivated() {
+
+        TransactionStatus status = transactionManager.getTransaction(new DefaultTransactionDefinition());
+
+        Subscriber subscriberIVR = subscriberDataService.findByNumber(1000000000L);
+        Set<Subscription> subscriptionsIVR = ( Set<Subscription> ) subscriberDataService.getDetachedField(subscriberIVR, "subscriptions");
+        for (Subscription subscriptionIVR : subscriptionsIVR) {
+            Assert.assertTrue(subscriptionIVR.getDeactivationReason().equals(DeactivationReason.WEEKLY_CALLS_NOT_ANSWERED));
+            Assert.assertTrue(subscriptionIVR.getStatus().equals(SubscriptionStatus.DEACTIVATED));
+        }
+
+        Subscriber subscriberMCTS = subscriberDataService.findByNumber(2000000000L);
+        Set<Subscription> subscriptionsMCTS = ( Set<Subscription> ) subscriberDataService.getDetachedField(subscriberMCTS, "subscriptions");
+        for (Subscription subscriptionMCTS : subscriptionsMCTS) {
+            Assert.assertTrue(subscriptionMCTS.getDeactivationReason().equals(DeactivationReason.WEEKLY_CALLS_NOT_ANSWERED));
+            Assert.assertTrue(subscriptionMCTS.getStatus().equals(SubscriptionStatus.DEACTIVATED));
+        }
+        transactionManager.commit(status);
+
+    }
+
+
+    public HttpDelete FormHttpRequest(Long msisdn) {
+        StringBuilder sb = new StringBuilder(deactivationRequest);
+        sb.append("?");
+        sb.append(String.format("msisdn=%s", msisdn.toString()));
+        HttpDelete httpRequest = new HttpDelete(sb.toString());
+        return httpRequest;
+    }
+
+     //Test deactivation of specific msisdn
+    @Test
+    public void testDeactivateSpecificValidMsisdn() throws IOException, InterruptedException, URISyntaxException {
+
+        HttpDelete httpRequestIVR = FormHttpRequest(1000000000L);
+        assertTrue(SimpleHttpClient.execHttpRequest(httpRequestIVR, HttpStatus.SC_OK, RequestBuilder.ADMIN_USERNAME, RequestBuilder.ADMIN_PASSWORD));
+
+        HttpDelete httpRequestMCTS = FormHttpRequest(2000000000L);
+        assertTrue(SimpleHttpClient.execHttpRequest(httpRequestMCTS, HttpStatus.SC_OK, RequestBuilder.ADMIN_USERNAME, RequestBuilder.ADMIN_PASSWORD));
+
+        testifSubscriberDectivated();
```

Comment:
probably want to pass in the phone number to this function to test the specific subscriber. not a deal breaker though.